### ISSUE TITLE
Using a representation of a low-rank symmetric matrix

### ISSUE THREAD

**Issue by sethaxen** [SebastianCallh/IterGP.jl#9]
Currently the `C` matrix is allocated as dense and updated in-place using rank-1 updates. This is fine when the data is not large (see #8) or when the rank is close to the size of the data, but otherwise it will be inefficient in terms of both storage and computation.

A better alternative would be to have an `AbstractMatrix` type that stores the rank-1 updates as either one long vector or a vector of vectors and then implement the necessary methods for this to act on other vectors or matrices.

**Comment by SebastianCallh** [comment]
You are absolutely right and that's a good suggestion. I have not pursued any optimizations yet since I don't fully trust the current implementation (https://github.com/SebastianCallh/IterGP.jl/issues/5).
